lib/auth.ts

```diff
@@ -133,6 +147,81 @@ export class Auth {
     return token;
   };
 
+  /**
+   * Perform a request against Kontist REST API
+   */
+  private request = async (path: string, method: HttpMethod, body?: string) => {
+    if (!this.token) {
+      throw new Error("User unauthorized");
+    }
+
+    const requestUrl = new URL(path, this.baseUrl).href;
+
+    const response = await fetch(requestUrl, {
+      method,
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.token.accessToken}`
+      },
+      body
+    });
+
+    return response.json();
+  };
+
+  /**
+   * Called periodically by `createMFAChallenge` until
+   * the challenge expires or its status is updated
+   */
+  private pollChallengeStatus = (
+    pendingChallenge: Challenge,
+    resolve: Function,
+    reject: Function
+  ) => async () => {
+    const challenge = await this.request(
+      `${MFA_CHALLENGE_PATH}/${pendingChallenge.id}`,
+      HttpMethod.GET
+    );
+
+    if (new Date(challenge.expiresAt) < new Date()) {
+      if (this.challengePollIntervalId) {
+        clearInterval(this.challengePollIntervalId);
+      }
+      reject(new Error("Challenge expired"));
+    } else if (challenge.status === ChallengeStatus.DENIED) {
+      if (this.challengePollIntervalId) {
+        clearInterval(this.challengePollIntervalId);
+      }
+      reject(new Error("Challenge denied"));
+    } else if (challenge.status === ChallengeStatus.VERIFIED) {
+      if (this.challengePollIntervalId) {
+        clearInterval(this.challengePollIntervalId);
+      }
+      const { token: confirmedToken } = await this.request(
```

Comment:
Are we are returning the plain token string here? Because `fetchToken` returns a `ClientOAuth2.Token` which e.g. has methods to refresh etc.
Does it make sense to align the output type?

Reply:
Absolutely right, I overlooked that, I have changed it here: 0184496a204b7b5b76369436cbfb94bf61a1c8be